### DISCUSSION TITLE
Eagerload additional relations if a particular relation is included in the query

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [spatie/laravel-query-builder#468] [Q&A] [answered]
So, I have a 2 relations added to my `allowedIncludes` method. 

Let's say for example 
```php
$query->allowedIncludes(['products','saleableProducts']);
```

If the request includes `saleable_products` to be included, I want to load additional child relations for `saleable_products` but not when the `products` relation is included. 

How do I go about doing this? Is there a way to check if the underlying query has a specific `include` included? 

Or can I do

```php
$query->allowedIncludes([
      'products',
      'saleableProducts.media',
      'saleableProducts.supplier',
      'saleableProducts.shops'
]);
```

and when `saleableProducts` is included in the request, then subsequently all the relations mentioned are automatically loaded?

**Comment by AlexVanderbist** [comment, marked as answer]
Hi @abishekrsrikaanth! I'm not sure if there's an easy way to conditionally load relationships based on _another relation being loaded in the same query_.

Under the hood the query builder simply adds the relations you request the an Eloquent query. So for `?include=products` we run `$query->with('products')`. This means there's no easy way to define a conditional include at the time of the request. There's ways around this but nothing supported by this package. 

For your second question:

>and when saleableProducts is included in the request, then subsequently all the relations mentioned are automatically loaded?

No, you have to manually add them to the URL as well: `GET /shops?include=saleableProducts.media,saleableProducts.supplier,saleableProducts.shops`.

Good luck!
